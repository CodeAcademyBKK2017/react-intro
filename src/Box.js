import React, { Component } from 'react';
import Row from './Row';

class Box extends Component {
  render() {
    return (
        <div>
            <Row/> <Row/> <Row/>
        </div>
    );
  }
}

export default Box;
