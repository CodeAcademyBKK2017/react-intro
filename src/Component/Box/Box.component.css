.Box{
    width: 300px;
    border: red solid;
    padding: 15px;
}
