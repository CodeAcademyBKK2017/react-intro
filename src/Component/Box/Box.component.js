import React, { Component } from 'react';
import Row from '../Row/Row.component';

class Box extends Component {
  render() {
    const my = this.props;
    return (
      <div>
        <Row cellData = {my.valueData[0]} clickHandler = {my.clickHandler}/>
        <Row cellData = {my.valueData[1]} clickHandler = {my.clickHandler}/>
        <Row cellData = {my.valueData[2]} clickHandler = {my.clickHandler}/>
      </div>
    );
  }
}

export default Box
