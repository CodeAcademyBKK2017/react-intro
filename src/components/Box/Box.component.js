import PropTypes from 'prop-types';
import React, {Component} from 'react';
import Row from '../Row/Row.component';

class Box extends Component {

  cellclickHandler (cellValue) {

    return () => {};

  }

  createRow = (rowData) => <Row cellData={rowData.items}
    key={rowData.id} cellclickHandler={this.cellclickHandler} />

  render () {

    const {boxData} = this.props;
    const rows = boxData.map(this.createRow);
    return (
      <div>
        {rows}
      </div>
    );

  }

}
Box.propTypes = {'boxData': PropTypes.array.isRequired};
Box.defaultProps = {'boxData': []};
export default Box;
