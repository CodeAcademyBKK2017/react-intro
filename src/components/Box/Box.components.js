import React, {Component} from 'react';
import PropTypes from 'prop-types';
import Row from '../Row/Row.components';
import noop from 'lodash/noop';
class Box extends Component {

  constructor () {
    super();
    this.cellClickHandler = this.cellClickHandler.bind(this);
  }

  // cellClickHandler = (cellValue) =>  // class preproties for es7 feature
  //   () => {
  //     console.log('cell clicked :: ', cellValue);
  //   }

  // class method : ES6
  cellClickHandler (cellValue) {
    return () => {
      console.log('cell clicked :: ', cellValue);
      
    };
  }
    
  // cellClickHandler (cellValue){ //class method
  //     console.log('cell clicked :: ', cellValue);
  // }
  render () {
    // this.props.rowData
    return (
      <div>
        <Row cellData={  this.props.rowData[0] } cellClickHandler={this.cellClickHandler}/> 
        <Row cellData={  this.props.rowData[1] } cellClickHandler={this.cellClickHandler}/> 
        <Row cellData={  this.props.rowData[2] } cellClickHandler={this.cellClickHandler}/>
      </div>
    );
  }
}


Box.propTypes = {
  cellClickHandler: PropTypes.func.isRequired,
  cellData: PropTypes.array.isRequired
};
Box.defaultProps = {
  cellData: [],
  cellClickHandler: noop
};

export default Box;
