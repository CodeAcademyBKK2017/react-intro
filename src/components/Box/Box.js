import React, { Component } from 'react';
import Row from '../Row/Row';

class Box extends Component {
  render() {
    return (
        <div>
            <Row/> <Row/> <Row/>
        </div>
    );
  }
}

export default Box;
