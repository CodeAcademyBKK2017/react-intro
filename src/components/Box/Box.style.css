.box {
    display: flex;
    border: 2px solid red;
    padding: 10vw;
    align-content: center;
    align-items: center;
}