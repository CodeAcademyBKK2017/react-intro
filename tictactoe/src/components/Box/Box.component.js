import React from 'react';

import Row from '../Row/Row.component'

class Box extends React.Component {
    cellClickHandler = (cellData) => () => {
        console.log('cell clicked', cellData);
    }
    render() {
        const rowLists = this.props.boxData.map((rowData, index) => <Row key={index} rowData={rowData} cellClickHandler={this.cellClickHandler} />);
        return (
            <div>
                {rowLists}
            </div>
        );
    }
}

export default Box;
