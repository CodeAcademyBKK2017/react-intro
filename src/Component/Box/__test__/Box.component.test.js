import React from 'react';
import Box from '../Box.component';
import renderer from 'react-test-renderer';

test('Box Snapshot Test',()=>{
  let valueData = [
    ['o','x','o'],
    ['x','o','o'],
    ['x','x','o']
  ];
  const tree = renderer.create(<Box data = {valueData} clickHandler = {jest.fn()}/>);
  expect(tree).toMatchSnapshot();
})
