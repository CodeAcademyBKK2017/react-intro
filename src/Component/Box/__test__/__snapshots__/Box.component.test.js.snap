// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Snapshot Test 1`] = `
<div>
  <div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#2196F3",
          "border": "0",
          "borderRadius": "2px",
          "boxShadow": " 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)",
          "color": "white",
          "display": "inline-block",
          "height": "80px",
          "justifyContent": "center",
          "margin": "3px",
          "textAlign": "center",
          "width": "80px",
        }
      }
    >
      <h1>
        o
      </h1>
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#2196F3",
          "border": "0",
          "borderRadius": "2px",
          "boxShadow": " 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)",
          "color": "white",
          "display": "inline-block",
          "height": "80px",
          "justifyContent": "center",
          "margin": "3px",
          "textAlign": "center",
          "width": "80px",
        }
      }
    >
      <h1>
        x
      </h1>
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#2196F3",
          "border": "0",
          "borderRadius": "2px",
          "boxShadow": " 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)",
          "color": "white",
          "display": "inline-block",
          "height": "80px",
          "justifyContent": "center",
          "margin": "3px",
          "textAlign": "center",
          "width": "80px",
        }
      }
    >
      <h1>
        o
      </h1>
    </div>
  </div>
  <div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#2196F3",
          "border": "0",
          "borderRadius": "2px",
          "boxShadow": " 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)",
          "color": "white",
          "display": "inline-block",
          "height": "80px",
          "justifyContent": "center",
          "margin": "3px",
          "textAlign": "center",
          "width": "80px",
        }
      }
    >
      <h1>
        x
      </h1>
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#2196F3",
          "border": "0",
          "borderRadius": "2px",
          "boxShadow": " 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)",
          "color": "white",
          "display": "inline-block",
          "height": "80px",
          "justifyContent": "center",
          "margin": "3px",
          "textAlign": "center",
          "width": "80px",
        }
      }
    >
      <h1>
        o
      </h1>
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#2196F3",
          "border": "0",
          "borderRadius": "2px",
          "boxShadow": " 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)",
          "color": "white",
          "display": "inline-block",
          "height": "80px",
          "justifyContent": "center",
          "margin": "3px",
          "textAlign": "center",
          "width": "80px",
        }
      }
    >
      <h1>
        o
      </h1>
    </div>
  </div>
  <div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#2196F3",
          "border": "0",
          "borderRadius": "2px",
          "boxShadow": " 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)",
          "color": "white",
          "display": "inline-block",
          "height": "80px",
          "justifyContent": "center",
          "margin": "3px",
          "textAlign": "center",
          "width": "80px",
        }
      }
    >
      <h1>
        x
      </h1>
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#2196F3",
          "border": "0",
          "borderRadius": "2px",
          "boxShadow": " 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)",
          "color": "white",
          "display": "inline-block",
          "height": "80px",
          "justifyContent": "center",
          "margin": "3px",
          "textAlign": "center",
          "width": "80px",
        }
      }
    >
      <h1>
        x
      </h1>
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "#2196F3",
          "border": "0",
          "borderRadius": "2px",
          "boxShadow": " 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)",
          "color": "white",
          "display": "inline-block",
          "height": "80px",
          "justifyContent": "center",
          "margin": "3px",
          "textAlign": "center",
          "width": "80px",
        }
      }
    >
      <h1>
        o
      </h1>
    </div>
  </div>
</div>
`;
